fetch/api/request/request-init-stream.any.js

```diff
@@ -2,50 +2,50 @@
 
 "use strict";
 
-async function assert_request(input, init) {
+function assert_request(test, input, init) {
   assert_throws(new TypeError(), () => new Request(input, init), "new Request()");
-  assert_throws(new TypeError(), async () => await fetch(input, init), "fetch()");
+  return promise_rejects_js(test, TypeError, fetch(input, init), "fetch()");
```

Comment:
How about keeping "async" and using

await promise_rejects(...)

here ?

Reply:
[`promise_rejects` is defined in terms of `assert_throws`](https://github.com/web-platform-tests/wpt/blob/a2c2030db3229967bcac856d06b1d8bf48c8de34/resources/testharness.js#L627-L631), and [`assert_throws` has previously been identified as prone to error](https://github.com/web-platform-tests/wpt/issues/11726).

Reply:
I'm fine with

await promise_rejects_js(...)

, too.

Reply:
If I understand correctly, you would prefer to "await" a Promise value as the final statement of an async function rather than return that value from a synchronous function. Is that correct?

Reply:
Yes, for this case.

Other functions in this file are async functions so making this an async function makes the code easier to understand, I think.

Reply:
Okay